CMakeLists.txt

```diff
@@ -50,203 +55,236 @@ link_directories(${UUID_LIBRARY})
 
 # LibClang library
 find_package(LibClang)
-if (HAVE_LIBCLANG)
-  include_directories(${LIBCLANG_INCLUDES})
+if(LIBCLANG_FOUND)
+  include_directories(${LIBCLANG_INCLUDE_DIRS})
   link_directories(${LIBCLANG_LIBRARY_DIR})
-endif (HAVE_LIBCLANG)
+endif(LIBCLANG_FOUND)
 
 # LLVM library
-find_package(LLVM)
-if (HAVE_LLVM)
+find_package(LLVM_config)
+if(LLVM_FOUND)
   include_directories(${LLVM_INCLUDE_DIR})
   link_directories(${LLVM_LIBRARY_DIR})
-endif (HAVE_LLVM)
+endif(LLVM_FOUND)
 
-set (LLVM_VERSION_MIN_REQUIRED "8.0.0")
+set(LLVM_VERSION_MIN_REQUIRED "13.0.0")
 
-if ( HAVE_LLVM AND
-     (NOT ( LLVM_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED )) AND
-     HAVE_LIBCLANG )
-  set ( ENABLE_CLANG_AS_LIB true )
+if(LLVM_FOUND
+   AND (NOT (LLVM_PACKAGE_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED))
+   AND LIBCLANG_FOUND)
+  set(ENABLE_CLANG_AS_LIB true)
 else()
-  if (NOT HAVE_LLVM)
-    message( WARNING "LLVM not detected" )
-  else(NOT HAVE_LLVM)
-    if (LLVM_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
-      message( WARNING "You are using an obsolete version of LLVM:\
-\n\t${LLVM_VERSION} detected.\t\
-Minimum required is ${LLVM_VERSION_MIN_REQUIRED}" )
-    else(LLVM_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
-      message( WARNING "LibClang not detected" )
-    endif (LLVM_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
-  endif (NOT HAVE_LLVM)
-  message( WARNING "Clang-as-a-library will be disabled" )
+  if(NOT LLVM_FOUND)
+    message(WARNING "LLVM not detected")
+  else(NOT LLVM_FOUND)
+    if(LLVM_PACKAGE_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
+      message(WARNING "You are using an obsolete version of LLVM:\
+\n\t${LLVM_PACKAGE_VERSION} detected.\t\
+Minimum required is ${LLVM_VERSION_MIN_REQUIRED}")
+    else(LLVM_PACKAGE_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
+      message(WARNING "LibClang not detected")
+    endif(LLVM_PACKAGE_VERSION VERSION_LESS LLVM_VERSION_MIN_REQUIRED)
+  endif(NOT LLVM_FOUND)
+  message(WARNING "Clang-as-a-library will be disabled")
 endif()
 
-set (VC_LIB_DEPS ${DL_LIBRARY}
-                 ${UUID_LIBRARY}
-                )
+set(VC_LIB_DEPS ${CMAKE_DL_LIBS} ${UUID_LIBRARY} stdc++)
```

Comment:
stdc++ should not be needed here, and it causes problems on platforms that use LLVM's own libc++ (which is linked as just -lc++ instead).

Reply:
Well, it seems required to build the tests, without it I'd get linking errors. 
I had attempted a fix which adds stdc++ on linux and c++ elsewhere. Let me know if it seems a proper one!